Heres' the webhook server for your bot.

mrjcleaver opened the pull request:
Hi Arjun,

So this webhook server for the summary bot works except that it doesn't actually summarize. For that I need help calling your code as I think it'll need refactoring to separate the summarization logic from the insertion of the result into a channel.

See the README.md file

Thanks,
   Martin

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added new Discord slash commands for summarizing channels, tagging content, and managing job queues.
  - Integrated enhanced webhook support and automated scheduling for seamless external service interactions.
  
- **Documentation**
  - Significantly updated the README and deployment guides with clear installation, usage, and troubleshooting instructions.
  
- **Refactor**
  - Improved error handling, logging, and performance optimizations for a more reliable bot experience.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mrjcleaver committed:
Initial support for an exposed webhook that runs a piece of code. This presently write into a hard coded channel id

mrjcleaver committed:
Added configurable webhook, tested against Zapier

mrjcleaver committed:
Added channels_to_summarize and altered channel_id to reflect that its a target

mrjcleaver committed:
gitignore to omit private files

ArjunSahlot commented:
Unfortunately, I'm not going to be merging this pull request, but, it seems like you have added some potentially useful changes. I'm just trying to be as objective as possible with my criticisms and definitely not trying to downplay your work. This pull request seems like a scrappy version you just got working for yourself and this won't fly on the production server. A lot of files are all over the place and the naming is not very specific. If you could describe the functionality of each file that'd be helpful in understanding the code. Looks like you added a lot of your experiments (call-webhook1, webhook-test, etc.), but, these experiments should stay on your local machine. There's multiple versions of a Flask apps littered around the code and overall it's very hard to understand what's going on. To me, it looks like you figured out how to host a server which listens for a request from a zapier action and sends a discord message in your server. This is a nice step but the real challenge comes in making this a feature for everyone, not just a hardcoded value. There's still a lot of work that needs to be done on this: creating a command for the user that would set this up, listening for requests and picking which server channel to send them in, etc. I'm sorry but this isn't in a mergable state right now without a lot more organizing and more progress towards making it usable by any number of people.

mrjcleaver committed:
Update webhook-server-example.json

the previous values invalidated

mrjcleaver committed:
Removed webhook-test

mrjcleaver committed:
Implement asynchronous LRU cache and integrate with OpenAI summarization for Discord bot

mrjcleaver committed:
Remove unused asynchronous LRU cache and related OpenAI summarization code

mrjcleaver committed:
Merge branch 'main' of https://github.com/mrjcleaver/summary_bot_opensource

mrjcleaver committed:
Added webhook-driven summaries. I'm working on documentation next.

mrjcleaver committed:
Update README and add TODO notes for code merger and refactoring

mrjcleaver committed:
Moved file

mrjcleaver committed:
Update CONTRIBUTORS.md

mrjcleaver committed:
Update CONTRIBUTORS.md

mrjcleaver committed:
Add APScheduler for webhook scheduling and update dependencies

mrjcleaver committed:
Merge branch 'main' of https://github.com/mrjcleaver/summary_bot_opensource

mrjcleaver committed:
renamed to scheduler

mrjcleaver committed:
Implement TaskWorker for job scheduling

mrjcleaver committed:
Add Google Drive integration and enhance summarization features

- Introduced Google Drive authentication and access via PyDrive.
- Updated dependencies in pyproject.toml to include PyDrive.
- Added new constants for message chunk size and context lookback days.
- Implemented asynchronous summarization for all channels with configurable time periods.
- Refactored existing functions to support improved logging and error handling.

mrjcleaver committed:
Add task list management, tagged channels functionality, and improve logging

mrjcleaver committed:
Add Fly.io configuration, Docker setup, and logging enhancements

mrjcleaver committed:
Add debugging support, environment checks, and update Fly.io configuration

mrjcleaver committed:
Refactor debug.py to use logging for debug messages and improve clarity

mrjcleaver committed:
Enhance debugging setup with 6tunnel for IPv6 support, update Dockerfile for dev dependencies, and improve logging in debug.py

mrjcleaver committed:
Switched from fly.io to local Docker instance, got remote IDE breakpoints working, switched to docker-compose

mrjcleaver committed:
Refactor debug.py for cleaner debugpy handling, enhance logging levels in main.py, and improve error handling in task_list.py; Checkpoint before refactoring summary.py

mrjcleaver committed:
Implemented changes suggested by Github's AI. Refactored Arun's summary.py

mrjcleaver committed:
Update bin/debugpy-connections-test.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mrjcleaver committed:
Update bin/fly-set-secrets.ps1

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mrjcleaver committed:
Refactor imports to use src. and invoke as src.main instead of main/  ; separate build from run in deploy script; use of bin/env for invoking scripts; turned off DEBUGPY_WAIT, tidied documentation

mrjcleaver committed:
Enhance Dockerfile and README; refactor deploy script for better logging; update main.py for improved debug handling; implement new chunking logic in ai_chunking2.py; add error handling in webhook processing; refine task management in task_list.py.

mrjcleaver committed:
Merge branch 'main' of https://github.com/mrjcleaver/summary_bot_opensource
